### ISSUE TITLE
client close should use conn.WriteCommand(COM_QUIT)

### ISSUE THREAD

**Issue by daiguadaidai** [siddontang/go-mysql#374]
see code:
```
conn, _ := client.Connect("127.0.0.1:3306", "root", "", "test")
conn.Close()
```
`conn.Close()` only close net. but not set qiut sign to `MySQL`

I Think you should use `conn.WriteCommand(COM_QUIT)`. Like:
file: `packet.conn.go`. line: `119`. func: `Close()`
now code:
```
func (c *Conn) Close() error {
	c.Sequence = 0
	if c.Conn != nil {
		return c.Conn.Close()
	}
	return nil
}
```

I thinkd code:
```
func (c *Conn) Close() error {
	return c.WriteCommand(COM_QUIT)
}
```

**Comment by siddontang** [comment]
yes, there is a problem, but we can't send QUIT here directly now because some places(like in handshake, binglog replication) which use Close function are not allowed to send QUIT. 

I will consider your suggestion, thank you.

**Comment by dveeden** [comment]
What about something like this:
```go
func (c *Conn) Quit() error {
	if err := c.writeCommand(COM_QUIT); err != nil {
                return err
        }
	return c.Close()
}
```

**Comment by dveeden** [comment]
@atercattus @siddontang @lance6716 what do you think about this?

Another option would be to check for a `_client_role` connection attribute with a value of `binary_log_listener`?

**Comment by lance6716** [comment]
@dveeden I prefer this library internally knows if COM_QUIT can be sent and send it automatically when Close(). It's friendly to user
